src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultHostDocumentFactory.cs

```diff
@@ -6,23 +6,34 @@
 using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor;
 using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
 using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer
 {
-    internal class DefaultHostDocumentFactory : HostDocumentFactory
+    internal class DefaultHostDocumentFactory : HostDocumentFactory, IDisposable
     {
+        private static readonly TimeSpan s_batchingTimeSpan = TimeSpan.FromMilliseconds(50);
```

Comment:
We may want to play with this number. One side effect of "setting output" is that it is the trigger to update the C#/HTML buffers on the client. The updating of the buffers is then also understood by various features to say "do I have the latest C#/HTML" before sending any request. Effectively 50ms MAY end up result in our synchronizer spinning for an extra 50ms before doing any feature. Now that being said, I use the word "may" here because technically there's a lot of other work going on prior to checking if the buffer is up-to-date. It's totally possible that things are slow enough to the point where when our synchronizer says "do I have the latest C#" it still does resulting in 0 perf impact.

Would you mind playing around with our [`DefaultLSPDocumentSynchronizer`](https://github.com/dotnet/razor-tooling/blob/a2011920e1ed4bde4ca38ed1efa2c0557f8a4fc4/src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentSynchronizer.cs#L45) and see if this change results in additional, unreasonable wait times for some of our features?

If it does, we could tweak this number to be significantly less given we're really just trying to minimize repeated work that takes too long

Reply:
The anecdata is in! From a debug build on my machine, running in the experimental hive... there is very little difference between anything. The old system had slightly more cache hits (ie, document was already at the right version) but we're talking 57% vs ~51%, so not much. With a 50ms timeout on the queue, the average duration to wait for a version was 61ms, for the old system, and for 10ms and 1ms timeouts, the average wait was 46ms.

With 1ms though, we had almost triple the amount of calls to `SetOutput` for output we already knew about (based on an input version check), though at 5 vs 13, the numbers are pretty low, I'm gonna err on the side of caution and go with 10ms for this.

Worth noting that there is _no_ difference in performance I could actually discern as far as how it felt to use the editor, typing C# and how long it took to get completion.

Reply:
Woooohoooo! Love the data and the follow up. Thank you for digging into this ❤️